app/test_split_integer.py

```diff
@@ -1,21 +1,38 @@
+import pytest
 from app.split_integer import split_integer
 
 
-def test_sum_of_the_parts_should_be_equal_to_value() -> None:
-    pass
-
-
-def test_should_split_into_equal_parts_when_value_divisible_by_parts() -> None:
-    pass
+class TestSplitIntegerFunction:
+    @pytest.mark.parametrize(
+        "entered_numbers,expected_result",
+        [
+            pytest.param(
+                (8, 1),
+                [8],
+                id="sum of the parts should be equal to value"
+            ),
+            pytest.param(
+                (32, 6),
+                [5, 5, 5, 5, 6, 6],
+                id="parts should be sorted when they are not equal"    
+            ),
+            pytest.param(
+                (1, 2),
+                [0, 1],
+                id="should add zeros when value is less than number of parts"
+            )
+        ]
+    )
+    def test_general_function_test(self, entered_numbers, expected_result) -> None:
+        result = split_integer(*entered_numbers)
+        assert result == expected_result
 
 
 def test_should_return_part_equals_to_value_when_split_into_one_part() -> None:
```

Comment:
The test `test_should_return_part_equals_to_value_when_split_into_one_part` seems to be testing a case not explicitly covered in the task description. The task specifies that the array should contain exactly `number_of_parts` integer elements, and the difference between the max and min number in the array should be <= 1. Consider revising this test to ensure it aligns with these requirements.